addons/payment_asiapay/models/payment_transaction.py

```diff
@@ -0,0 +1,133 @@
+import logging
+
+from werkzeug import urls
+
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+from odoo.tools.float_utils import float_compare
+
+from .const import CURRENCY_MAPPING
+from odoo.addons.payment_asiapay import utils as payment_asiapay_utils
+from odoo.addons.payment_asiapay.controllers.main import AsiaPayController
+
+_logger = logging.getLogger(__name__)
+
+class PaymentTransaction(models.Model):
+    _inherit = "payment.transaction"
+
+    def _get_specific_rendering_values(self, processing_values):
+        """Override of payment to return asiapay rendering values.
+
+        Note: self.ensure_one() from `_get_processing_values`
+
+        :param dict processing_values: The generic and specific processing values of the transaction
+        :return: The dict of acquirer-specific processing values
+        :rtype: dict
+        """
+        res = super()._get_specific_rendering_values(processing_values)
+        if self.provider != "asiapay":
+            return res
+
+        base_url = self.acquirer_id.get_base_url()
+        rendering_values = {
+            "merchantId": self.acquirer_id.asiapay_merchant_id,
+            "amount": self.currency_id._convert(self.amount, self.acquirer_id.asiapay_currecy_id, self.company_id, fields.Date.today()),
+            "orderRef": self.reference,
+            "currCode": CURRENCY_MAPPING.get(self.acquirer_id.asiapay_currecy_id.name),
+            "mpsMode": "NIL",
+            "successUrl": urls.url_join(base_url, AsiaPayController._return_url),
+            "failUrl": urls.url_join(base_url, AsiaPayController._return_url),
+            "cancelUrl": urls.url_join(base_url, AsiaPayController._return_url),
+            "payType": "N",
+            "lang": "E",
+            # "payMethod": "CC",
```

Comment:
remove comment

Reply:
Currently I allowed all payment method. However should we limit the payment method choice?